extensions/metrics/src/metrics.component.ts

```diff
@@ -28,18 +28,24 @@ export class MetricsComponent implements Component {
     @inject(CoreBindings.APPLICATION_INSTANCE)
     private application: Application,
     @config()
-    options: MetricsOptions = DEFAULT_METRICS_OPTIONS,
+    metricsConfig: MetricsConfig = {},
   ) {
-    if (!options.defaultMetrics || !options.defaultMetrics?.disabled) {
+    const options: MetricsOptions = {
+      ...DEFAULT_METRICS_OPTIONS,
+      ...metricsConfig,
+    };
+    if (options.defaultMetrics && !options.defaultMetrics.disabled) {
```

Comment:
It can be simplified as:

```ts
if (!options.defaultMetrics?.disabled) {
```

Reply:
its actually not the same if `options.pushGateway` is `undefined` it will be true whereas `options.pushGateway && !options.pushGateway.disabled` is false becasue `!undefined === true`

Reply:
`if (options.defaultMetrics?.disabled !== true)` is probably the simplest.

Reply:
this does not work in case of the `pushGateway` which is undefined and we dont want to add the `MetricsPushObserver` in that case